src/video_core/command_classes/codecs/codec.cpp

```diff
@@ -59,6 +60,36 @@ Codec::~Codec() {
     av_buffer_unref(&av_gpu_decoder);
 }
 
+#ifdef LIBVA_FOUND
+// List all the currently loaded Linux modules
+static std::vector<std::string> ListLinuxKernelModules() {
+    std::vector<std::string> modules{};
+    auto module_listing = fopen("/proc/modules", "rt");
```

Comment:
```suggestion
    using FILEPtr = std::unique_ptr<FILE, decltype(&std::fclose)>;
    auto module_listing = FILEPtr{std::fopen("/proc/modules", "rt"), std::fclose};
```

Which allows the closing to be handled automatically